### PR TITLE
Closes #1541 xASL_io_SaveNifti: earlier removing redundant .mat sidecar

### LINKED ISSUE #1541: Redundant .mat sidecar warning
### Description

xASL_io_SaveNifti: Minor bug in which the order of removing a redundant .mat sidecar and some other checks needed to be shifted. We first did some checks that would reload a newly created NIfTI, which would then catch a residual .mat file with the wrong number of volumes. Now with the new order this warning is not incorrectly given anymore.

Relevant as this warning can confuse us when testing new flavors that deal differently with motion estimation/correction.

### Tasks

- [x] xASL_io_SaveNifti: earlier removing redundant .mat sidecar

### Release notes

Bugfix xASL_io_SaveNifti: earlier removing redundant .mat sidecar

### PR DESCRIPTION
**Pull request by HenkMutsaerts**
### Linked issue

Closes #1541

### CODE REVIEW

File: External/SPMmodified/xASL/xASL_io_SaveNifti.m [hunk 1]
```diff
@@ -177,6 +179,22 @@ function xASL_io_SaveNifti(pathOrigNifti, pathNewNifti, imNew, nBits, bGZip, cha
...
 newNifti.dat(:,:,:,:,:) = imNew;
 
+
+%% 4. Remove redundant .mat orientation files
+if size(imNew,4)==1
+    xASL_delete(tempMat);
+end
+
+if exist(tempMat, 'file')
+	tmpMat = load(tempMat);
```
Comment (jan-petr) [inline External/SPMmodified/xASL/xASL_io_SaveNifti.m]: It is all fine. But we might want to rename `tmpMat` to something else as it looks a bit like a typo in `tempMat`.

File: External/SPMmodified/xASL/xASL_io_SaveNifti.m [hunk 2]
```diff
@@ -177,6 +179,22 @@ function xASL_io_SaveNifti(pathOrigNifti, pathNewNifti, imNew, nBits, bGZip, cha
 
 newNifti.dat(:,:,:,:,:) = imNew;
 
+
```
Comment (jan-petr) [inline External/SPMmodified/xASL/xASL_io_SaveNifti.m:182]: We can update date to 2023 at line 39.

**Review by jan-petr — CHANGES_REQUESTED** [review]
Minor change.

File: External/SPMmodified/xASL/xASL_io_SaveNifti.m (hunk 2 above)
Comment (HenkMutsaerts) [reply External/SPMmodified/xASL/xASL_io_SaveNifti.m:182]: Done

File: External/SPMmodified/xASL/xASL_io_SaveNifti.m (hunk 1 above)
Comment (HenkMutsaerts) [reply External/SPMmodified/xASL/xASL_io_SaveNifti.m]: I'm probably not the one who wrote this code, knowing how I dislike such abbreviations ;)
Renamed to motionMat

**Review by jan-petr — APPROVED** [review]
OK.

**Pull request merged**